chain-signatures/node/src/protocol/presignature.rs

```diff
@@ -650,6 +651,54 @@ impl PresignatureManager {
 
         messages
     }
+
+    pub fn execute(
+        state: &RunningState,
+        active: &Participants,
+        protocol_cfg: &ProtocolConfig,
+    ) -> tokio::task::JoinHandle<()> {
+        let triple_manager = state.triple_manager.clone();
+        let presignature_manager = state.presignature_manager.clone();
+        let active = active.clone();
+        let protocol_cfg = protocol_cfg.clone();
+        let pk = state.public_key.clone();
+        let sk_share = state.private_share.clone();
+        let messages = state.messages.clone();
+
+        tokio::task::spawn(async move {
+            let mut presignature_manager = presignature_manager.write().await;
```

Comment:
The benefits of spawning multiple tasks for concurrent execution might be reduced if the first thing we do is to acquire a write lock on a shared object across all tasks. Especially if we keep the lock until it is all done, which I think is the case here.

Reply:
good find! This will definitely contend with the signature task. I left this in for now, but I meant to separate out the contention of the presignature_manager in this PR as well similar to how `TripleManager` had internalized it's locks. I will make it less contentious in another PR or maybe in the same PR